lib/jsdom/living/documents.js

```diff
@@ -0,0 +1,31 @@
+"use strict";
+const XMLDocument = require("../living/generated/XMLDocument");
+const Document = require("../living/generated/Document");
+
+exports.createWrapper = (globalObject, options = {}, type = options.parsingMode || null) => {
```

Comment:
I'm unsure about the defaulting of type to options.parsingMode. It seems weird that you have code to synchronize the two both in the XMLDocumentImpl constructor, and here. The function signature gives the false impression that they can be decoupled, e.g. you could pass a parsingMode of HTML with a type of XML.

I'm not sure what the best function signature would be, but maybe something like `(globalObject, options = {}, { useParsingModeBasedSubclass = false } = {})`? Or some shorter variant of that.